lib/Server.js

```diff
@@ -130,12 +130,13 @@ class Server {
         next(err);
       });
 
-      return new Promise(resolve => {
+      const listenPromise = new Promise(resolve => {
         this.httpServer = this.server.listen(this.port, () => {
           console.log(`Server listening at http://${this.hostname}:${this.port}`);
           resolve();
         });
       });
+      return Promise.all([listenPromise, readyPromise]);
```

Comment:
The reason for having `Server.start()` not resolve its promise until all servers are up and running is so that the HTTP request made by `createBundle()` doesn't time out while the RN packager and webpack are still traversing the file system for their dependencies. On slow machines this can take quite a while. With this change, the HTTP request doesn't get made until both packagers are ready to receive requests.